app/src/test/java/org/simple/clinic/instantsearch/InstantSearchEffectHandlerTest.kt

```diff
@@ -335,4 +339,29 @@ class InstantSearchEffectHandlerTest {
     verify(uiActions).openQrCodeScanner()
     verifyNoMoreInteractions(uiActions)
   }
+
+  @Test
+  fun `when check if patient has an existing NHID effect is received, then check check if patient has an existing NHID`() {
```

Comment:
We have two branches added to the effect handler, but only one test. Are you not following the TDD process?

Reply:
I am not handling the `ShowNHIDErrorDialog` effect here. I have written tests for the Update function where its being dispatched from. I handle this effect in the next commit.

Reply:
Oh, you mean a test for asserting `PatientDoesNotHaveAnExistingNHID` event. I will do this.

Reply:
Yeah, I think a bigger concern is missing this test but having this branch present in the production code. This implies that you aren't following our TDD process, because otherwise you'd have added the test first before writing the production code.

Please make it a point to follow the process.